server-spi-private/src/main/java/org/keycloak/services/clientpolicy/ClientPoliciesUtil.java

```
@@ -0,0 +1,747 @@
+            throw new ClientPolicyException("proposed client policy name duplicated.", "NA");
+        }
+
+        // construct updating policies from existing policies and proposed policies
+        ClientPoliciesRepresentation updatingPoliciesRep = new ClientPoliciesRepresentation();
+        updatingPoliciesRep.setPolicies(new ArrayList<>());
+        List<ClientPolicyRepresentation> updatingPoliciesList = updatingPoliciesRep.getPolicies();
+
+        // add existing builtin policies to updating policies
+        List<ClientPolicyRepresentation> existingPoliciesList = existingPoliciesRep.getPolicies();
+        if (existingPoliciesList != null && !existingPoliciesList.isEmpty()) {
+            existingPoliciesList.stream().filter(i->i.isBuiltin()).forEach(i->updatingPoliciesList.add(i));
+        }
+
+        for (ClientPolicyRepresentation proposedPolicyRep : proposedPoliciesRep.getPolicies()) {
+            if (proposedPolicyRep.getName() == null) {
+                continue;
+            }
+
+            // ignore newly proposed builtin policy because builtin policy cannot added/deleted/modified.
+            Boolean enabled = (proposedPolicyRep.isEnable() != null) ? proposedPolicyRep.isEnable() : Boolean.FALSE;
+            if (proposedPolicyRep.isBuiltin() != null && proposedPolicyRep.isBuiltin()) {
+                // only enable field of the existing builtin policy can be overridden.
+                if (updatingPoliciesList.stream().anyMatch(i->i.getName().equals(proposedPolicyRep.getName()))) {
+                    updatingPoliciesList.stream().filter(i->i.getName().equals(proposedPolicyRep.getName())).forEach(i->i.setEnable(enabled));
+                    continue;
+                }
+                logger.warnv("{0} builtin policy included in proposed policies. Just ignore it. policy name = {1}", LOGMSG_PREFIX, proposedPolicyRep.getName());
```

Again, my vote is to throw the error instead of just display the warning

---

@mposolda I see. I would like to fix codes to throw the error as you previously proposed.